src/sql/workbench/contrib/notebook/browser/markdownToolbarActions.ts

```diff
@@ -473,39 +494,61 @@ function getColumnOffsetForSelection(type: MarkdownButtonType, nothingSelected:
 	}
 }
 
-export class TogglePreviewAction extends ToggleableAction {
+export class ToggleMarkdownViewAction extends Action {
+	constructor(
+		id: string,
+		label: string,
+		cssClass: string,
+		tooltip: string,
+		private _cellModel: ICellModel,
+	) {
+		super(id, label, cssClass);
+		this._tooltip = tooltip;
+	}
 
-	private static readonly previewShowLabel = localize('previewShowLabel', "Show Preview");
-	private static readonly previewHideLabel = localize('previewHideLabel', "Hide Preview");
-	private static readonly baseClass = 'codicon';
-	private static readonly previewShowCssClass = 'split-toggle-on';
-	private static readonly previewHideCssClass = 'split-toggle-off';
-	private static readonly maskedIconClass = 'masked-icon';
+	public async run(context: ICellModel): Promise<boolean> {
+		context = this._cellModel;
+		context.showPreview = false;
+		context.showTextView = false;
+		return true;
+	}
+}
 
+export class ToggleSplitViewAction extends Action {
 	constructor(
-		id: string, toggleTooltip: boolean, showPreview: boolean
+		id: string,
+		label: string,
+		cssClass: string,
+		tooltip: string,
+		private _cellModel: ICellModel,
 	) {
-		super(id, {
-			baseClass: TogglePreviewAction.baseClass,
-			toggleOffLabel: TogglePreviewAction.previewShowLabel,
-			toggleOffClass: TogglePreviewAction.previewShowCssClass,
-			toggleOnLabel: TogglePreviewAction.previewHideLabel,
-			toggleOnClass: TogglePreviewAction.previewHideCssClass,
-			maskedIconClass: TogglePreviewAction.maskedIconClass,
-			shouldToggleTooltip: toggleTooltip,
-			isOn: showPreview
-		});
+		super(id, label, cssClass);
+		this._tooltip = tooltip;
 	}
-
-	public get previewMode(): boolean {
-		return this.state.isOn;
+	public async run(context: ICellModel): Promise<boolean> {
+		context = this._cellModel;
+		context.showPreview = true;
+		return true;
 	}
-	public set previewMode(value: boolean) {
-		this.toggle(value);
+}
+export class ToggleTextViewAction extends Action {
+	constructor(
+		id: string,
+		label: string,
+		cssClass: string,
+		tooltip: string,
+		private _cellModel: ICellModel,
+	) {
+		super(id, label, cssClass);
+		this._tooltip = tooltip;
 	}
-	public async run(context: any): Promise<boolean> {
-		this.previewMode = !this.previewMode;
-		context.cellModel.showPreview = this.previewMode;
+	public async run(context: ICellModel): Promise<boolean> {
+		// Higher up, the remaining need to be set to unselected.
```

Comment:
What does this mean?

Reply:
I need a way to set the other two buttons to unselected by removing any existing active class, then allow a button to mark itself as active by adding the active class. So far, I do not have a solution.